### PR TITLE
drivers: sensors: Add support for ds18s20

### PR DESCRIPTION
**Pull request by fgihl**
Added support for the older ds18s20 inside the (newer) ds18b20.

### CODE REVIEW

File: dts/bindings/sensor/maxim,ds18b20.yaml [hunk 1]
```diff
@@ -14,3 +14,8 @@ properties:
     description: |
         Resolution of the temperature reading in bit.
         Valid configurations are from 9 to 12 bits.
+  ds18s20:
```
Comment (decsny) [inline dts/bindings/sensor/maxim,ds18b20.yaml:17]: instead of a boolean property, I think it would be more accurate to add another compatible

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add a node with the new compatible to `tests/drivers/build_all/sensor/w1.dtsi`

File: drivers/sensor/maxim/ds18b20/ds18b20.c [hunk 2]
```diff
@@ -25,15 +27,46 @@ LOG_MODULE_REGISTER(DS18B20, CONFIG_SENSOR_LOG_LEVEL);
...
+ */
+static const char *device_str(const struct device *dev)
+{
+	const struct ds18b20_config *cfg = dev->config;
+
+	if (cfg->chip == type_ds18s20)
+		return "DS18S20";
+	else
+		return "DS18B20";
+}
```
[the remark this answers is not in the document]
Comment (fgihl) [reply drivers/sensor/maxim/ds18b20/ds18b20.c]: Fixed

File: drivers/sensor/maxim/ds18b20/ds18b20.c [hunk 3]
```diff
@@ -25,15 +27,32 @@ LOG_MODULE_REGISTER(DS18B20, CONFIG_SENSOR_LOG_LEVEL);
...
-	int16_t temp = sys_get_le16(temp_raw);
+	const struct ds18b20_config *cfg = dev->config;
 
-	val->val1 = temp / 16;
-	val->val2 = (temp % 16) * 1000000 / 16;
+	return cfg->chip == type_ds18s20;
+}
+
+static inline void ds18b20_temperature_from_raw(
+						const struct device *dev,
```
Comment (str4t0m) [inline drivers/sensor/maxim/ds18b20/ds18b20.c:43]: please remove the new lline before the first argument.

File: drivers/sensor/maxim/ds18b20/ds18b20.c [hunk 4]
```diff
@@ -25,15 +27,32 @@ LOG_MODULE_REGISTER(DS18B20, CONFIG_SENSOR_LOG_LEVEL);
...
+static const uint16_t measure_wait_ds18s20_ms = { 750 };
+
+static bool is_ds18s20(const struct device *dev)
 {
-	int16_t temp = sys_get_le16(temp_raw);
+	const struct ds18b20_config *cfg = dev->config;
 
-	val->val1 = temp / 16;
-	val->val2 = (temp % 16) * 1000000 / 16;
+	return cfg->chip == type_ds18s20;
```
Comment (str4t0m) [inline drivers/sensor/maxim/ds18b20/ds18b20.c:39]: Not sure this function really brings a benefit, it only makes a simple check.
I think is would be more clear to just do that check in the if conditions directly.

File: tests/drivers/build_all/sensor/w1.dtsi [hunk 5]
```diff
@@ -12,3 +12,10 @@ test_w1_ds18b20: ds18b20 {
 	resolution = <12>;
 	status = "okay";
 };
+
+test_w1_ds18s20: ds18s20 {
+	compatible = "maxim,ds18s20";
+	family-code = <0x10>;
+    resolution = <9>;
+	status = "okay";
```
Comment (str4t0m) [inline tests/drivers/build_all/sensor/w1.dtsi:20]: Please fix the white space issue.

**Review by str4t0m — CHANGES_REQUESTED** [review]

File: dts/bindings/sensor/maxim,ds18s20.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2024, Fredrik Gihl
+# SPDX-License-Identifier: Apache-2.0
+
+description: Maxim 1-Wire ds18s20 temperature sensor
+
+compatible: "maxim,ds18s20"
+
+include: maxim,ds18b20.yaml
```
[the remark this answers is not in the document]
Comment (fgihl) [reply dts/bindings/sensor/maxim,ds18s20.yaml:8]: Yes, better to separate them.

**Comment by fgihl** [comment]
rebased to solve conflict in release note file

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by str4t0m — APPROVED** [review]

**Pull request merged**